### PR TITLE
[MI-1413] Add support for Policy Interaction Kind

### PR DESCRIPTION
**Pull request by r4vi**
Issue number: MI-1413

### Description of change

Adds a new Interaction kind called 'policy', the frontend will be updated to add support for creating these policy interactions.

### Checklist

* [x] ~Have any relevant search models been updated?~
* [x] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [x] ~Have any relevant select-/prefetch-related field lists in the views and search apps been updated?~
* [x] ~Has the admin site been updated (for new models, fields etc.)?~
* [x] ~Has the README been updated (if needed)?~

### CODE REVIEW

**Comment by r4vi** [comment]
@jim68000 @reupen I think this is all we need to do on the backend

File: datahub/interaction/models.py [hunk 1]
```diff
@@ -74,6 +74,7 @@ class Interaction(BaseModel):
     KINDS = Choices(
         ('interaction', 'Interaction'),
         ('service_delivery', 'Service delivery'),
+        ('policy', 'Policy Interaction'),
```
Comment (reupen) [inline datahub/interaction/models.py]: Minor thing but we generally use sentence case for titles etc. (so it would be `Policy interaction` in this case).

**Comment by r4vi** [comment]
@reupen thanks for the comments, I've made a fixup commit if you think it's fine I'll rebase

File: datahub/interaction/migrations/0008_add_interaction_kind.py [hunk 2]
```diff
@@ -15,7 +15,11 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='interaction',
             name='kind',
-            field=models.CharField(choices=[('interaction', 'Interaction'), ('service_delivery', 'Service delivery')], default='interaction', max_length=255),
+            field=models.CharField(choices=[
```
Comment (reupen) [inline datahub/interaction/migrations/0008_add_interaction_kind.py]: Sorry, this migration was squashed into `0001_squashed_0019_rename_default_permissions` but the old migrations haven't been removed yet. So it'll need to be changed in `0001_squashed_0019_rename_default_permissions`. If you run `./manage.py makemigrations` afterwards, it should say there are no changes.

**Comment by reupen** [comment]
You might want to review the validation rules on InteractionSerializer as well as they only handle the two existing interaction kinds. But you could do that separately if you want.

**Comment by r4vi** [comment]
@reupen thanks for the advice, I'll do the changes in the serializer in this PR too

**Comment by r4vi** [comment]
@reupen 
* we want to treat `policy` interactions the same as `interaction` for now so I've updated the serializer to match `interaction`. Added a new InRule and tests for it
* I've updated the right migration file now :ok_hand:

File: datahub/core/test/test_validators.py [hunk 3]
```diff
@@ -194,6 +195,17 @@ def test_equals_rule(data, field, test_value, res):
     assert condition(combiner) == res
 
 
+@pytest.mark.parametrize('data,field,test_value,res', (
+    ({'colour': 'red'}, 'colour', ['red', 'green'], True),
+    ({'colour': 'red'}, 'colour', ['blue', 'green'], False),
+))
+def test_in_rule(data, field, test_value, res):
+    """Tests ValidationCondition for various cases."""
```
Comment (reupen) [inline datahub/core/test/test_validators.py]: The docstring is wrong here (but so are the ones in the preceding functions :grimacing: )

**Comment by reupen** [comment]
@r4vi That looks good. But would suggest adding some tests using that kind with the views and/or paramet(e)rising some of the existing tests.

**Comment by r4vi** [comment]
@reupen https://github.com/uktrade/data-hub-leeloo/pull/840/commits/9f3815901fd97473e86f21fdd13e4246a6408aef paramertises some of the existing tests

**Pull request merged**